code/modules/mob/living/silicon/robot/inventory.dm

```diff
@@ -1,82 +1,233 @@
-	O.cyborg_unequip(src)
 
+	switch(module_num)
+		if(1)
+			if(!(disabled_modules & BORG_MODULE_ALL_DISABLED))
+				inv1.icon_state = "inv1"
+		if(2)
+			if(!(disabled_modules & BORG_MODULE_TWO_DISABLED))
+				inv2.icon_state = "inv2"
+		if(3)
+			if(!(disabled_modules & BORG_MODULE_THREE_DISABLED))
+				inv3.icon_state = "inv3"
+
+	if(I.item_flags & DROPDEL)
+		I.item_flags &= ~DROPDEL //we shouldn't HAVE things with DROPDEL_1 in our modules, but better safe than runtiming horribly
+		
+	held_items[module_num] = null
+	I.cyborg_unequip(src)
+	I.forceMove(module) //Return item to module so it appears in its contents, so it can be taken out again.
+
+	observer_screen_update(I, FALSE)
 	hud_used.update_robot_modules_display()
-	return 1
+	return TRUE
+
+/**
+  * Breaks the slot number, changing the icon.
+  *
+  * Arguments
+  * * module_num - the slot number being repaired.
+  */
+/mob/living/silicon/robot/proc/break_cyborg_slot(module_num)
+	if(is_invalid_module_number(module_num, TRUE))
+		return FALSE
 
-/mob/living/silicon/robot/proc/activate_module(obj/item/O)
-	. = FALSE
-	if(!(O in module.modules))
-		return
-	if(activated(O))
-		to_chat(src, "<span class='warning'>That module is already activated.</span>")
-		return
-	if(!held_items[1])
-		held_items[1] = O
-		O.screen_loc = inv1.screen_loc
-		. = TRUE
-	else if(!held_items[2])
-		held_items[2] = O
-		O.screen_loc = inv2.screen_loc
-		. = TRUE
-	else if(!held_items[3])
-		held_items[3] = O
-		O.screen_loc = inv3.screen_loc
-		. = TRUE
-	else
-		to_chat(src, "<span class='warning'>You need to disable a module first!</span>")
-	if(.)
-		O.equipped(src, ITEM_SLOT_HANDS)
-		O.mouse_opacity = initial(O.mouse_opacity)
-		O.layer = ABOVE_HUD_LAYER
-		O.plane = ABOVE_HUD_PLANE
-		observer_screen_update(O,TRUE)
-		O.forceMove(src)
-		if(istype(O, /obj/item/borg/sight))
-			var/obj/item/borg/sight/S = O
-			sight_mode |= S.sight_mode
-			update_sight()
+	if(held_items[module_num]) //If there's a held item, unequip it first.
+		if(!unequip_module_from_slot(held_items[module_num], module_num)) //If we fail to unequip it, then don't continue
+			return FALSE
+
+	switch(module_num)
+		if(1)
+			if(disabled_modules & BORG_MODULE_ALL_DISABLED)
+				return FALSE
+		
+			inv1.icon_state = "inv1 +b"
+			disabled_modules |= BORG_MODULE_ALL_DISABLED
 
+		if(2)
+			if(disabled_modules & BORG_MODULE_TWO_DISABLED)
+				return FALSE
+		
+			inv2.icon_state = "inv2 +b"
+			disabled_modules |= BORG_MODULE_TWO_DISABLED
+
+		if(3)
+			if(disabled_modules & BORG_MODULE_THREE_DISABLED)
+				return FALSE
+		
+			inv3.icon_state = "inv3 +b"
+			disabled_modules |= BORG_MODULE_THREE_DISABLED
+
+	playsound(loc, 'sound/machines/warning-buzzer.ogg', 50, TRUE, TRUE)
+	audible_message("<span class='warning'>[src] sounds an alarm! \"SYSTEM ERROR: Module [module_num] OFFLINE.\"</span>", self_message = null)
+	to_chat(src, "<span class='userdanger'>SYSTEM ERROR: Module [module_num] OFFLINE.</span>")
```

Comment:
```
 					audible_message("<span class='warning'>[src] sounds an alarm! \"CRITICAL ERROR: All modules OFFLINE.\"</span>")
					to_chat(src, "<span class='userdanger'>CRITICAL ERROR: All modules OFFLINE.</span>")
					playsound(loc, 'sound/machines/warning-buzzer.ogg', 75, TRUE, TRUE)
```

preserve the old functionality of these getting louder with each one that gets disabled / a message saying everything is disabled when everything is... well, disabled.